app/views/pages/home.html.haml

```diff
@@ -6,7 +6,7 @@
   %p
     - if signed_in?
       %li= link_to t('log_out'), destroy_user_session_path, method: :delete, class: "btn btn-primary btn-sm mt-10"
-      %li= link_to t('profile', name: current_user.name),users_profile_path(current_user.id) ,method: :get, class: "btn btn-primary btn-sm mt-10"
+      %li= link_to t('profile', name: current_user.name), users_profile_path(current_user.profile.id), class: "btn btn-primary btn-sm mt-10"
```

Comment:
hmmm... I see that only looged in users can see and edit their profiles... So, let's have single route users_profile_path and find current_user profile inside controller. 
(have in your routes `resource :profile, only: %i[show edit update]`)